src/spark_rapids_ml/clustering.py

```diff
@@ -43,6 +43,8 @@
 from spark_rapids_ml.params import HasFeaturesCols, _CumlClass, _CumlParams
 from spark_rapids_ml.utils import get_logger
 
+P = TypeVar("P", bound=_CumlParams)
```

Comment:
since this is repeated in several files, should it be in a common file?

Reply:
Yeah, thought about that, but it looks most people define locally per file.

Reply:
Hmm, looks like Spark actually uses a mix of local-defines and imports, so it's not consistent.
```
$ find python/pyspark -name "*.py" | xargs grep TypeVar | grep '"P"'
python/pyspark/ml/feature.py:P = TypeVar("P", bound=Params)
python/pyspark/ml/param/__init__.py:P = TypeVar("P", bound="Params")
$ find python/pyspark -name "*.py" | xargs grep "import P$"
python/pyspark/ml/base.py:from pyspark.ml.param import P
python/pyspark/ml/tree.py:    from pyspark.ml._typing import P
```

Reply:
Went ahead and updated with the `import P` method, since it's less repeated code.  Also updated `tree.py`, which wasn't failing any checks, just to be consistent.